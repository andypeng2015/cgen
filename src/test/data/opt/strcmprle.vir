module foo

export function sw $foo(l %s1, l %s2) {
@start:
  jmp @next
@next:
  %c1 = ld.b %s1
  %c2 = ld.b %s2
  jmp @check1
@check1:
  %nz = ne.b %c1, 0_b
  br %nz, @check2, @diff
@check2:
  %eq = eq.b %c1, %c2
  br %eq, @inc, @diff
@inc:
  %s1 = add.l %s1, 1_l
  %s2 = add.l %s2, 1_l
  jmp @next
@diff:
  %c1 = ld.b %s1
  %c2 = ld.b %s2
  %d = sub.b %c1, %c2
  %r = sext.w %d
  ret %r
}
