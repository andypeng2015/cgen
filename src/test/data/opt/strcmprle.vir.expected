module foo

export function sw $foo(l %s1, l %s2) {
@0:
  jmp @1(%s2, %s1)
@1(%s2.1, %s1.1):
  %c1.1 = ld.b %s1.1 ; @3
  %c2.1 = ld.b %s2.1 ; @4
  %0 = ne.b %c1.1, 0x0_b ; @16
  br %0, @5, @6
@5:
  %3 = eq.b %c1.1, %c2.1 ; @19
  br %3, @8, @6
@8:
  %5 = add.l %s1.1, 0x1_l ; @21
  %4 = add.l %s2.1, 0x1_l ; @20
  jmp @1(%4, %5)
@6:
  %2 = sub.b %c1.1, %c2.1 ; @18
  %1 = sext.w %2 ; @17
  ret %1
}
